packages/cursorless-engine/src/languages/TreeSitterQuery/queryPredicateOperators.ts

```diff
@@ -150,6 +150,25 @@ class ShrinkToMatch extends QueryPredicateOperator<ShrinkToMatch> {
   }
 }
 
+/**
+ * A predicate operator that modifies the range of the match by trimming trailing whitespace,
+ * similar to the javascript trimEnd function.
+ */
+class TrimEnd extends QueryPredicateOperator<TrimEnd> {
```

Comment:
I'd be tempted to just have a bidirectional trim. Then you could leverage our `trimRange` utility function. Curious why yours is so much shorter than that one tho 🤔

Reply:
> I'd be tempted to just have a bidirectional trim.

I guess the question is whether we are going to want `trim`, `trimStart`, and `trimEnd`. I opted for `trimEnd` on the grounds that it was precisely what I needed, and that we could add `trim` and `trimStart` if we need that behavior. (In this case `trim` and `trimEnd` are practically identical, but I preferred the precise semantics for documentation and clarity.) I chose the name to match the built-in javascript function it calls.

> Curious why yours is so much shorter than that one tho 

Why does `trimRange` use regexes instead of calling the javascript built-in `trim`/`trimStart`/`trimEnd`?

Reply:
WDYT about having a util something like this? (untested!)

```typescript
export function adjustPosition(doc: TextDocument, pos: Position, by: number): Position {
  const offset = document.offsetAt(pos);
  return document.positionAt(offset + by);
}
```

Then the final part of `trimRange` shrinks to:

```typescript
return new Range(
    adjustPosition(editor.document, range.start, leadingWhitespace.length),
    adjustPosition(editor.document, range.end, -trailingWhitespace.length),
  );
```

which is (IMHO) clear and pretty.

And the end of this new function is:

```typescript
    nodeInfo.range = new Range(
      range.start,
      adjustPosition(document, range.end, trimmed.length - text.length),
    );
    return true;
```

which is simple enough that it hardly seems worth attempting to DRY out.

[Go proverb: A little copying is better than a little dependency.](https://www.youtube.com/watch?v=PAAkCSZUG1c&t=568s)

Reply:
`trimRange` also has a latent bug when the range is entirely whitespace. I'll go ahead and send a PR cleaning up a bunch of this.

Reply:
lol. it's also dead code.

Reply:
OK, sent a PR to cull the dead code, and added a commit making this function a bit clearer.